dd-compile-policy/Makefile

```diff
@@ -37,69 +72,142 @@ build: flatbuffers $(CXXOPTS_HEADER) $(POLICY_BFBS) $(POLICY_BFBS_HEADER) $(OUT)
 else
 	cd $(BUILD_DIR) && cmake -DCMAKE_CXX_FLAGS="-Os -fno-exceptions -fno-rtti -fvisibility=hidden -Wno-stringop-overflow" -DCMAKE_BUILD_TYPE=MinSizeRel -DFLATBUFFERS_BUILD_TESTS=OFF -DFLATBUFFERS_BUILD_FLATC=OFF -DFLATBUFFERS_BUILD_FLATHASH=OFF ..
 endif
 	cd $(BUILD_DIR) && cmake --build . --target flatbuffers
 
 flatbuffers: $(BUILD_DIR)/libflatbuffers.a
+endif
 
 # Generate Flatbuffers binary schema (bfbs)
-$(POLICY_BFBS): $(SCHEMA_PATH)
+$(POLICY_BFBS): $(SCHEMA_PATH) flatbuffers
+	$(MKDIR) schema
+ifeq ($(DETECTED_OS),Windows)
+	$(BUILD_DIR)/MinSizeRel/flatc.exe --binary --force-defaults --schema -o schema $(SCHEMA_PATH)
+else
 	flatc --binary --force-defaults --schema -o schema $(SCHEMA_PATH)
+endif
 
-# Generate Flatbuffers header file from binary schema
+# Generate C header file from binary schema
+# On Unix: use xxd tool to convert binary to hex array  
+# On Windows: use PowerShell script (xxd not available)
 $(POLICY_BFBS_HEADER): $(POLICY_BFBS)
+ifeq ($(DETECTED_OS),Windows)
+	# Create temporary PowerShell script to convert binary to C header
+	# This mimics the behavior of 'xxd -i' on Unix systems
+	@echo '$$bytes = Get-Content "$(POLICY_BFBS)" -Encoding Byte' > convert_binary.ps1
+	@echo '$$hexArray = $$bytes | ForEach-Object { "0x{0:x2}" -f $$_ }' >> convert_binary.ps1
+	@echo '$$hexString = $$hexArray -join ", "' >> convert_binary.ps1  
+	@echo '$$header = "unsigned char schema_policy_bfbs[] = { " + $$hexString + " };"' >> convert_binary.ps1
+	@echo '$$length = "unsigned int schema_policy_bfbs_len = " + $$bytes.Length + ";"' >> convert_binary.ps1
+	@echo '$$header + [System.Environment]::NewLine + $$length | Out-File "$(POLICY_BFBS_HEADER)" -Encoding ascii' >> convert_binary.ps1
+	powershell -ExecutionPolicy Bypass -File convert_binary.ps1
+	powershell -Command "Remove-Item convert_binary.ps1"
+else
 	xxd -i $< > $@
+endif
 
 # Build output binary
 $(OUT): $(SRC) flatbuffers $(CXXOPTS_HEADER)
-	mkdir -p $(MAKEFILE_DIR)/bin
-ifeq ($(shell uname), Linux)
+	$(MKDIR) $(MAKEFILE_DIR)/bin
+ifeq ($(DETECTED_OS),Windows)
+	$(CXX) $(CXXFLAGS) $(INCLUDES) $(SRC) $(LIBS) /Fe$@ $(LDFLAGS)
+else
+ifeq ($(DETECTED_OS),Linux)
 	$(CXX) $(CXXFLAGS) $(INCLUDES) $(SRC) $(LIBS) $(LDFLAGS) -o $@
 else
 	$(CXX) $(CXXFLAGS) $(INCLUDES) $(SRC) $(LIBS) -o $@
 endif
+endif
 
 strip_binary:
+ifneq ($(DETECTED_OS),Windows)
 ifeq (, $(shell which $(STRIP)))
 	@echo "strip not found, skipping binary strip"
 else
 	$(STRIP) $(OUT)
 endif
+endif
 
 
 # Verify by unmarshalling compiled policy to JSON
 verify_json:
 ifndef BIN
 	$(error You must specify BIN=<path_to_bin_file>)
 endif
+ifeq ($(DETECTED_OS),Windows)
+	$(BUILD_DIR)/MinSizeRel/flatc.exe --json --strict-json --raw-binary -o . $(SCHEMA_PATH) -- $(BIN)
```

Comment:
I think we could just have a variable FLATC for the entire file that's `$(BUILD_DIR)/MinSizeRel/flatc.exe` on Windows and `flatc` on other OS